code/game/objects/structures/mineral_doors.dm

```diff
@@ -81,6 +86,9 @@
 			else
 				SwitchState()
 	else if(ismecha(user))
+		if(lock == 2)
```

Comment:
Can we get the lock states (0,1, and 2) a predefine?
Like LOCK_BROKEN LOCK_UNLOCKED and LOCK_LOCKED? Just to make maintenance easier

Other than that (and the other requests), totally solid idea, thanks for the work